### PR TITLE
[4.4] Remove unused cassiopeia inline event handler

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
The current Cassiopeia version has an inline onload event handler for the fontscheme.current css file. The inline event handler produces a warning in strict CSP mode and is redundant, as the lazy-loading functionality (that was the purpose of the handler) is handled in the template.js and trigger by the rel="lazy-stylesheet" attribute.

### Testing Instructions
* Configure the "roboto (local)" font as font in cassiopeia, verify that it's loaded by the browser using the browser dev tools.
* Configure a strict ("script-src: 'self' 'unsafe-eval'") report-only script CSP using the Joomla HTTP Header plugin:

Bildschirmfoto 2024-06-01 um 14 20 13

### Actual result BEFORE applying this Pull Request
* CSS file loaded
* Error message about forbidden unsafe-inline JS calls, caused by an "onload" handler of the mentioned CSS file in the browser console shown

Bildschirmfoto 2024-06-01 um 14 21 00

### Expected result AFTER applying this Pull Request
* CSS file loaded
* Error message about unsafe-inline calls about the onload handler is gone

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
Please provide more test instructions and what is expected to be seen in the generated source

Maybe I am looking at this wrong but before this PR I see

`<link href="/media/templates/site/cassiopeia/css/global/fonts-local_roboto.min.css?8eef51" rel="stylesheet" media="all" onload="this.media='all'">`

With the PR applied I see

`<link href="/media/templates/site/cassiopeia/css/global/fonts-local_roboto.min.css?447e02" rel="stylesheet" media="all" onload="this.media='all'">`

**Comment by brianteeman** [comment]
Also does the same change need to be made in 
`templates\cassiopeia\component.php`
`templates\cassiopeia\error.php`

File: templates/cassiopeia/index.php [hunk 1]
```diff
@@ -49,7 +49,7 @@
         $this->getPreloadManager()->preconnect('https://fonts.googleapis.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preconnect('https://fonts.gstatic.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preload($paramsFontScheme, ['as' => 'style', 'crossorigin' => 'anonymous']);
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'', 'crossorigin' => 'anonymous']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/index.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```

the media part is not needed anymore and if left will never load the fonts for the normal use ie displaying the site on a screen.

Comment (dgrammatiko) [reply templates/cassiopeia/index.php]: Also please change the same parts at line 73

File: templates/cassiopeia/component.php [hunk 2]
```diff
@@ -32,7 +32,7 @@
         $this->getPreloadManager()->preconnect('https://fonts.googleapis.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preconnect('https://fonts.gstatic.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preload($paramsFontScheme, ['as' => 'style', 'crossorigin' => 'anonymous']);
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'', 'crossorigin' => 'anonymous']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/component.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```

File: templates/cassiopeia/component.php [hunk 3]
```diff
@@ -41,7 +41,7 @@
 			--cassiopeia-font-weight-headings: 700;';
         }
     } else {
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/component.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['rel' => 'lazy-stylesheet']);
```

File: templates/cassiopeia/error.php [hunk 4]
```diff
@@ -45,7 +45,7 @@
         $this->getPreloadManager()->preconnect('https://fonts.googleapis.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preconnect('https://fonts.gstatic.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preload($paramsFontScheme, ['as' => 'style', 'crossorigin' => 'anonymous']);
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'', 'crossorigin' => 'anonymous']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/error.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```

File: templates/cassiopeia/error.php [hunk 5]
```diff
@@ -54,7 +54,7 @@
 			--cassiopeia-font-weight-headings: 700;';
         }
     } else {
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/error.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['rel' => 'lazy-stylesheet']);
```

File: templates/cassiopeia/index.php [hunk 6]
```diff
@@ -70,7 +70,7 @@
     		--cassiopeia-font-weight-headings: 700;';
         }
     } else {
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/index.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['rel' => 'lazy-stylesheet']);
```

File: templates/cassiopeia/offline.php [hunk 7]
```diff
@@ -39,7 +39,7 @@
         $this->getPreloadManager()->preconnect('https://fonts.googleapis.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preconnect('https://fonts.gstatic.com/', ['crossorigin' => 'anonymous']);
         $this->getPreloadManager()->preload($paramsFontScheme, ['as' => 'style', 'crossorigin' => 'anonymous']);
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'', 'crossorigin' => 'anonymous']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/offline.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, [], ['rel' => 'lazy-stylesheet', 'crossorigin' => 'anonymous']);
```

File: templates/cassiopeia/offline.php [hunk 8]
```diff
@@ -48,7 +48,7 @@
 			--cassiopeia-font-weight-headings: 700;';
         }
     } else {
-        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet', 'onload' => 'this.media=\'all\'']);
+        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['media' => 'print', 'rel' => 'lazy-stylesheet']);
```
Comment (dgrammatiko) [inline templates/cassiopeia/offline.php]:
```suggestion
        $wa->registerAndUseStyle('fontscheme.current', $paramsFontScheme, ['version' => 'auto'], ['rel' => 'lazy-stylesheet']);
```

**Review by dgrammatiko — CHANGES_REQUESTED** [review]
@SniperSister please remove all the `'media' => 'print'`

**Comment by SniperSister** [comment]
Lesson learned: don't commit while listening to JDay presentations ;) 

Sorry guys, now all required changes (removal of inline handler, removal of unused media attribute) have been performed in all relevant files and locations.

**Comment by brianteeman** [comment]
As stated before please provide more detailed test instructions. How can a tester replicate the error etc

**Comment by SniperSister** [comment]
> As stated before please provide more detailed test instructions. How can a tester replicate the error etc

Done!

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f3bbd2d7eb6f814d8effb6327f4db26fbdcfcfe7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43586">issues.joomla.org/tracker/joomla-cms/43586</a>.</sub>

**Comment by richard67** [comment]
@SniperSister Shouldn't it be fixed in 4.4-dev, too (or only there and later be merged up to 5.1-dev)? If it's a bug fix we should to it also in 4.4-dev, and if it's a new feature (what I don't think) it should to into 5.2-dev.

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on f3bbd2d7eb6f814d8effb6327f4db26fbdcfcfe7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43586">issues.joomla.org/tracker/joomla-cms/43586</a>.</sub>

**Comment by SniperSister** [comment]
> Shouldn't it be fixed in 4.4-dev, too (or only there and later be merged up to 5.1-dev)?

@richard67 fair comment. I have rebased the PR as suggested!

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**